assets/js/modules/thank-with-google/components/common/ColorRadio.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_THANK_WITH_GOOGLE } from '../../datastore/constants';
+import ImageRadio from '../../../../components/ImageRadio';
+import { getColorThemes } from '../../util/settings';
+const { useSelect, useDispatch } = Data;
+
+export default function ColorRadio() {
+	const { setColorTheme } = useDispatch( MODULES_THANK_WITH_GOOGLE );
+
+	const currentColor = useSelect( ( select ) =>
+		select( MODULES_THANK_WITH_GOOGLE ).getColorTheme()
+	);
+
+	const onChange = useCallback(
+		( { target } = {} ) => {
+			const { value: color = 'blue' } = target || {};
+			setColorTheme( color );
+		},
+		[ setColorTheme ]
+	);
+
+	const colors = getColorThemes()?.map(
+		( { colorThemeID, name, svg: SVG } ) => (
+			<ImageRadio
+				key={ `${ currentColor }::${ colorThemeID }` }
```

Comment:
`currentColor` being part of the key causes re-rendering of these components (and accessibility issues with focus) because the key's value changes between every React render.

Because the `key` here changes every time `currentColor` changes (and `currentColor` changes whenever a new colour is selected with the arrow keys):

![CleanShot 2022-07-27 at 20 20 54](https://user-images.githubusercontent.com/90871/181355827-c03ba313-8d67-4173-93c8-340fa68cb9a0.gif)

Fortunately, the fix is straightforward:

```suggestion
				key={ `${ name }::${ colorThemeID }` }
```

or even just
```suggestion
				key={ colorThemeID }
```

works. When tested locally this then behaves properly with fewer DOM re-renders and no keyboard issues:

![CleanShot 2022-07-27 at 20 25 46](https://user-images.githubusercontent.com/90871/181356128-86911753-b7a3-46a4-a92d-7633a7c15b34.gif)

Reply:
Yes, you are right. Updated.